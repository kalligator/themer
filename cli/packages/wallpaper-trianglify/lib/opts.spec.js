const { getSizes, getVariances, getSeed } = require('./opts');

describe('@themerdev/wallpaper-trianglify options', () => {
  it('should return proper defaults if none provided', () => {
    expect(getSizes()).toMatchSnapshot();
    expect(getVariances()).toMatchSnapshot();
  });
  it('should parse a single resolution option', () => {
    expect(getSizes('200x200')).toEqual([{ h: 200, w: 200 }]);
  });
  it('should parse multiple resolution options', () => {
    expect(getSizes(['100x200', '300x400'])).toEqual([
      { w: 100, h: 200 },
      { w: 300, h: 400 },
    ]);
  });
  it('should throw when a malformed resolution option is given', () => {
    expect(() => getSizes('100xFoo')).toThrow();
  });
  it('should parse a single variance option', () => {
    expect(getVariances('0.61')).toEqual([0.61]);
  });
  it('should parse multiple variance options', () => {
    expect(getVariances(['0.333', '0.1', '1'])).toEqual([0.333, 0.1, 1]);
  });
  it('should throw when an invalid variance option is given', () => {
    expect(() => getVariances('1.1')).toThrow();
    expect(() => getVariances('-1')).toThrow();
    expect(() => getVariances('foo')).toThrow();
  });
});
