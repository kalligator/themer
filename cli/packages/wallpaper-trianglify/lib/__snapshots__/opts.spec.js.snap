// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@themerdev/wallpaper-trianglify options should return proper defaults if none provided 1`] = `
Array [
  Object {
    "h": 1800,
    "w": 2880,
  },
  Object {
    "h": 1334,
    "w": 750,
  },
]
`;

exports[`@themerdev/wallpaper-trianglify options should return proper defaults if none provided 2`] = `
Array [
  0.75,
]
`;
